cmd/paths.go

```diff
@@ -256,77 +256,124 @@ func pathsFetchCmd(a *appState) *cobra.Command {
-				ReferenceName: "refs/heads/main",
-			}); err != nil {
-				return err
-			}
+			combinations := permutations(chains)
 
-			// Try to fetch path info for each configured chain that has canonical chain/path info in the GH repo
-			for _, srcChain := range a.Config.Chains {
-				for _, dstChain := range a.Config.Chains {
+			client := github.NewClient(nil)
+			gh_ctx := context.Background()
 
-					// Add paths to rly config from {localRepo}/interchain/chaind-id/
-					localPathsDir := path.Join(localRepo, "interchain", srcChain.ChainID())
+			overwrite, _ := cmd.Flags().GetBool(flagOverwriteConfig)
 
-					dir := path.Clean(localPathsDir)
-					files, err := ioutil.ReadDir(dir)
-					if err != nil {
-						fmt.Fprintf(cmd.ErrOrStderr(), "path info does not exist for chain: %s. Consider adding its info to %s. Error: %v\n", srcChain.ChainID(), path.Join(PATHSURL, "interchain"), err)
-						break
-					}
+			for _, pthName := range combinations {
 
-					// For each path file, check that the dst is also a configured chain in the relayers config
-					for _, f := range files {
-						pth := filepath.Join(dir, f.Name())
-
-						if f.IsDir() {
-							fmt.Fprintf(cmd.ErrOrStderr(), "directory at %s, skipping...\n", pth)
-							continue
-						}
-
-						byt, err := os.ReadFile(pth)
-						if err != nil {
-							cleanupDir(localRepo)
-							return fmt.Errorf("failed to read file %s: %w", pth, err)
-						}
-
-						p := &relayer.Path{}
-						if err = json.Unmarshal(byt, p); err != nil {
-							cleanupDir(localRepo)
-							return fmt.Errorf("failed to unmarshal file %s: %w", pth, err)
-						}
-
-						if p.Dst.ChainID == dstChain.ChainID() {
-							pthName := strings.Split(f.Name(), ".")[0]
-							if err = a.Config.AddPath(pthName, p); err != nil {
-								return fmt.Errorf("failed to add path %s: %w", pth, err)
-							}
-
-							fmt.Fprintf(cmd.ErrOrStderr(), "added path %s...\n", pthName)
-						}
-					}
+				_, exist := a.Config.Paths[pthName]
+				if exist && !overwrite {
+					fmt.Printf("skipping:  %s already exists in config, use -o to overwrite (clears filters) \n", pthName)
+					continue
+				}
 
-					if err := a.OverwriteConfig(a.Config); err != nil {
-						return err
+				// TDOD: Don't use github api. Potentially use: https://github.com/eco-stake/cosmos-directory once they integrate IBC data into restAPI. This will avoid rate limits.
+				fileName := pthName + ".json"
+				reg_path := filepath.Join("_IBC", fileName)
+				r, err := client.Repositories.DownloadContents(gh_ctx, "cosmos", "chain-registry", reg_path, nil)
+				if err != nil {
+					if _, ok := err.(*github.RateLimitError); ok {
+						return fmt.Errorf("hit github rate limit ERR: %w", err)
 					}
+					fmt.Printf("not found:  path %s not found in repo 'cosmos/chain-registry', folder '_IBC'\n", pthName)
+					continue
+				}
+
+				bytes, err := ioutil.ReadAll(r)
+				if err != nil {
+					fmt.Printf("error reading response body: %v", err)
```

Comment:
Returning the error is appropriate (otherwise we will continue with missing or garbage data).